lib/phoenix/socket/transport.ex

```diff
@@ -353,6 +353,51 @@ defmodule Phoenix.Socket.Transport do
     end
   end
 
+  @doc """
+  Checks the Websocket subprotocol request header against the allowed subprotocol.
+
+  Should be called by transports before connecting when appropriate.
+  If the sec-websocket-protocol header matches the allowed subprotocol,
+  it will put sec-websocket-protocol response header and return the given connection.
+  If no sec-websocket-protocol header was sent it will return the given connection.
+
+  Otherwise a 403 Forbidden response will be sent and the connection halted.
+  It is a noop if the connection has been halted.
+  """
+  def check_origin(conn, subprotocol)
+
+  def check_subprotocol(%Plug.Conn{halted: true} = conn, _subprotocol), do: conn
+  def check_subprotocol(conn, nil), do: conn
+
+  def check_subprotocol(conn, subprotocol) do
+    import Plug.Conn
+    subprotocols = get_req_header(conn, "sec-websocket-protocol")
```

Comment:
Shouldn't we split this on comma and remove spaces? You can use this function for help: https://hexdocs.pm/plug/Plug.Conn.Utils.html#list/1